pkg/cmd/certregenerationcontroller/certregenerationcontroller.go

```diff
@@ -0,0 +1,302 @@
+package certregenerationcontroller
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/klog"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configeversionedclient "github.com/openshift/client-go/config/clientset/versioned"
+	configexternalinformers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/library-go/pkg/operator/certrotation"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/genericoperatorclient"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/certrotationcontroller"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+)
+
+type Options struct {
+	Workers        int
+	KubeconfigPath string
+	TlsServerName  string
+
+	LeaderelectionLeaseDuration time.Duration
+	LeaderelectionRenewDeadline time.Duration
+	LeaderelectionRetryPeriod   time.Duration
+
+	restConfig   *restclient.Config
+	kubeClient   kubernetes.Interface
+	configClient configeversionedclient.Interface
+}
+
+func NewCertRegenerationControllerCommand() *cobra.Command {
+	o := &Options{
+		Workers:        1,
+		KubeconfigPath: "",
+		TlsServerName:  "",
+
+		LeaderelectionLeaseDuration: 60 * time.Second,
+		LeaderelectionRenewDeadline: 35 * time.Second,
+		LeaderelectionRetryPeriod:   10 * time.Second,
+	}
+
+	cmd := &cobra.Command{
+		Use: "cert-regeneration-controller",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate()
+			if err != nil {
+				return err
+			}
+
+			err = o.Complete()
+			if err != nil {
+				return err
+			}
+
+			err = o.Run()
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	cmd.PersistentFlags().IntVarP(&o.Workers, "workers", "", o.Workers, "Number of workers to run")
```

Comment:
are we ever going to support more than 1 worker?

Reply:
no, I think that pattern can be removed for 1 key

Reply:
done